tests/test_routing.py

```diff
@@ -482,3 +482,25 @@ def handler2():
     _, handler, params = router.get(f"/api/3/hello_world/{uri}", "GET")
     assert handler() == "handler2"
     assert params == {"version": "3", "foo": uri}
+
+
+@pytest.mark.parametrize("uri", ("a-random-path", "a/random/path"))
+def test_identical_path_routes_with_different_methods_similar_urls(uri):
+    def handler1():
+        return "handler1"
+
+    # test root level path with different methods
+    router = Router()
+    router.add("/constant/<foo:path>/story", handler1, methods=["GET", "OPTIONS"])
+    router.add("/constant/<foo:path>/tracker/events", handler1, methods=["PUT"])
+    router.add("/constant/<foo:path>/tracker/events", handler1, methods=["POST"])
```

Comment:
Can we add `handler2` and `handler3` and then do asserts to make sure each is routed to the expected location? Should be an example to copy.

Reply:
will do

Reply:
done (or should I assert the `route` as well?)

Reply:
perfect